mllib/src/test/scala/org/apache/spark/ml/fpm/FPGrowthSuite.scala

```diff
@@ -0,0 +1,120 @@
+      val model = new FPGrowth().setMinSupport(0.8).fit(intData)
+      val generatedRules = model.setMinConfidence(0.8).getAssociationRules
+      val expectedRules = spark.createDataFrame(Seq(
+        (Array("2"), Array("1"), 1.0),
+        (Array("1"), Array("2"), 1.0)
+      )).toDF("antecedent", "consequent", "confidence")
+        .withColumn("antecedent", col("antecedent").cast(ArrayType(dt)))
+        .withColumn("consequent", col("consequent").cast(ArrayType(dt)))
+
+      assert(expectedRules.sort("antecedent").rdd.collect().sameElements(
+        generatedRules.sort("antecedent").rdd.collect()))
+      val transformed = model.transform(intData)
+      assert(transformed.count() == 3)
+    }
+  }
+
+  test("FPGrowth getFreqItems") {
+    val model = new FPGrowth().setMinSupport(0.8).fit(dataset)
+    val expectedFreq = spark.createDataFrame(Seq(
+      (Array("1"), 3L),
+      (Array("2"), 3L),
+      (Array("1", "2"), 3L)
+    )).toDF("items", "freq")
+    val freqItems = model.getFreqItemsets
+    assert(freqItems.sort("items").rdd.collect()
+      .sameElements(expectedFreq.sort("items").rdd.collect()))
+  }
+
+  test("FPGrowth get Association Rules") {
+    val model = new FPGrowth().setMinSupport(0.8).fit(dataset)
+    val expectedRules = spark.createDataFrame(Seq(
+      (Array("2"), Array("1"), 1.0),
+      (Array("1"), Array("2"), 1.0)
+    )).toDF("antecedent", "consequent", "confidence")
+    val associationRules = model.getAssociationRules
+
+    assert(associationRules.sort("antecedent").rdd.collect()
+      .sameElements(expectedRules.sort("antecedent").rdd.collect()))
+  }
+
+  test("FPGrowth parameter check") {
+    val fpGrowth = new FPGrowth().setMinSupport(0.4567)
+    val model = fpGrowth.fit(dataset)
+                  .setMinConfidence(0.5678)
+    assert(fpGrowth.getMinSupport === 0.4567)
+    assert(model.getMinConfidence === 0.5678)
+  }
+
+  test("read/write") {
+    def checkModelData(model: FPGrowthModel, model2: FPGrowthModel): Unit = {
+      assert(model.freqItemsets.sort("items").collect() ===
+        model2.freqItemsets.sort("items").collect())
+    }
+    val fPGrowth = new FPGrowth()
+    testEstimatorAndModelReadWrite(
+      fPGrowth, dataset, FPGrowthSuite.allParamSettings, checkModelData)
+  }
+
+}
+
+object FPGrowthSuite {
+
+  def getFPGrowthData(spark: SparkSession): DataFrame = {
+    spark.createDataFrame(Seq(
+      (0, Array("1", "2", "3", "5")),
+      (0, Array("1", "2", "3", "6")),
+      (0, Array("1", "2", "7"))
+    )).toDF("id", "features")
+  }
+
+  /**
+   * Mapping from all Params to valid settings which differ from the defaults.
+   * This is useful for tests which need to exercise all Params, such as save/load.
+   * This excludes input columns to simplify some tests.
+   */
+  val allParamSettings: Map[String, Any] = Map(
+    "minSupport" -> 0.321,
+    "minConfidence" -> 0.456,
+    "numPartitions" -> 5,
+    "featuresCol" -> "features",
```

Comment:
Exclude input columns here; it complicates the standard tests if we don't use default values for inputs.